### ISSUE TITLE
SIngle click event on MarkerClusterer stopped working

### ISSUE THREAD

**Issue by aritrasyd** [JustFly1984/react-google-maps-api#3022]
## Your Environment

os: mac windows

node --version 16.14.2

react version 17.0.2

@react-google-maps/api version 2.1.1

The single click on the cluster does not work any more with the lastest package build using the version 2.1.1 or higher. The single click still works if I use the previously installed node modules. Using the below codesandbox.com url, the same issue can be reproduced using version 2.8.1 onwards. Now, user needs to double click to zoom on the cluster.

## Basic implementation of incorrect behavior in codesandbox.com

https://codesandbox.io/s/marker-clusterer-api-forked-ex4tmj?file=/src/Map2.js:552-562

**Comment by kevindoole** [comment]
Ditto! I perused the source but couldn't find a solution. I noticed a couple possible angles that may be clues, but didn't confirm:

- the `MarkerClusterer` `options` prop seems to have some overlap with it's props in the docs. Eg. `imagePath` is given is an option, but it's also in the list of props
- in @react-google-maps/marker-clusterer, they have an option called `onClusterClick` -- perhaps `onClick` is mixed up with `onClusterClick`
- the `zoomOnClick` functionality is also not working

**Comment by pedromedley** [comment]
Hey there! Just submitted a PR to fix this issue. The problem is that the event listeners in the ClusterIcon should be added to the div and not via google.maps.event.addListener according to the deprecation message of the google.maps.event. addDomListener.

**Comment by JustFly1984** [comment]
I've released 2.10.3. fixed memory leaks

**Comment by pedromedley** [comment]
@JustFly1984 now there is a problem where in the new event handler functions, the this is referencing to the div and not to ClusterIcon. I will submit a new PR to fix this and include 2 additional changes:
1. the ClusterIconStyle.className is not applied to the cluster div on the first draw, only after the ClusterIcon is re-added -> will assign the className to the div in the  show() function instead on the add()
2. adding the width and the height properties to the img so it will match the ClusterIcon.width and ClusterIcon.height -> this will allow to provide images with higher resolution so they can look crispier

**Comment by pedromedley** [comment]
@JustFly1984 just created a new PR

**Comment by JustFly1984** [comment]
Can we close an issue now? Please test latest release 2.11.8

**Comment by kevindoole** [comment]
Confirmed working for me too -- thanks!!! ❤️ 🏆

**Comment by aritrasyd** [comment]
Thanks for the fix. The latest version worked for me ❤️

**Comment by JagadeeshGeerla** [comment]
Thanks for the fix. Works for me as well.

**Comment by pedromedley** [comment]
Also here it is working fine...
